docs/ref/forms/api.txt

```diff
@@ -952,6 +959,31 @@ Attributes of ``BoundField``
 
         <label for="myFIELD">...</label><input id="myFIELD" type="text" name="my_field" required>
 
+.. attribute:: BoundField.initial
+
+    Use :attr:`BoundField.initial` to retrieve initial data for a form field.
+    It retrieves the value from :attr:`Form.initial`, or :attr:`Field.initial`
+    if not present, and then evaluates if a callable. The return value is
+    cached::
+
+        >>> import uuid
+        >>> class UUIDCommentForm(CommentForm):
+        ...     identifier = forms.UUIDField(initial=uuid.uuid4)
+        >>> f = UUIDCommentForm()
+        >>> print(f['identifier'].initial)
+        9c50c0f6-966f-475e-9e1f-fb591e5dadaf
+        >>> print(f['identifier'].initial)
+        9c50c0f6-966f-475e-9e1f-fb591e5dadaf
+
+    Using :attr:`BoundField.initial` is recommended over
```

Comment:
I'm not sure we need the alternative showing that the non cached version changes. 🤔

Otherwise maybe chop the words but add the code examples to the above section with a code comment?

Reply:
As yet another alternative, this could be included under `Form.get_initial_for_field()`'s docs, but it would require defining `UUIDCommentForm` in two places.

Reply:
PS - I also like these `get_initial_for_field()` code examples because they show concretely how the `BoundField.initial` is simpler. :)

Reply:
I resolved this by putting the `get_initial_for_field()` code in the `get_initial_for_field()` section, and limiting mention of `get_initial_for_field()` in `BoundField.initial`'s to a brief sentence at the end. This minimizes the extraneous info when reading the "happy path" of the docs, while still acknowledging `get_initial_for_field()` for people that have code that can be improved, and wouldn't otherwise come across this info.